src/main/java/edu/harvard/iq/dataverse/util/bagit/BagGenerator.java

```diff
@@ -204,7 +205,15 @@ public boolean generateBag(OutputStream outputStream) throws Exception {
         // The oremapObject is javax.json.JsonObject and we need com.google.gson.JsonObject for the aggregation object
         aggregation = (JsonObject) new JsonParser().parse(oremapObject.getJsonObject(JsonLDTerm.ore("describes").getLabel()).toString());
 
-        bagID = aggregation.get("@id").getAsString() + "v."
+        String pidUrlString = aggregation.get("@id").getAsString();
+        String pidString=pidUrlString;
+        //ToDo - put this conversion in GlobalId

```

Comment:
Is there a reason to not do this todo now (seems small enough)?

Reply:
Makes sense. FWIW: I knew #8665 was coming (as well as #8674) when I wrote this . With #8665 merged, I've gone ahead and moved the logic to the GlobalID class (updated for #8665) (and added a ToDo there for further cleanup that I hope to do in #8674).